enhancements/cluster-api/installing-cluster-api-components-in-ocp.md

```diff
@@ -0,0 +1,290 @@
+The Cluster Version Operator (CVO) deploys two Second Level Operators, the OpenShift-specific cluster-capi-operator, and the upstream cluster-api-operator.
+At build time we generate and embed specific Custom Resources and the manifests ConfigMap defined for the upstream operator within the Openshift-specific operator,
+we template them, apply them, and let the upstream operator pick those up, act on them by unpacking the locally available ConfgMap, from which the CAPI providers manifests get unnested and deployed.
+
+**Development and Payload Stability**.
+The development of this complex system, as previously described, has various steps to it and happens across various repositories, where multiple PRs need to merge before a provider change/bump can actually completely land on an OpenShift payload as a feature.
+The problem with this is that the various PRs that it takes to land a provider rebase, for example, cannot be merged atomically, but will land at separate stages once they are reviewed, approved, their CI is green and the merge bot decides it's time for them to be merged.
+This can lead to payload breakages when a provider image build is not backward compatible with the previous version's manifests (Deployment/CRDs/RBACs), which in the current stage are deployed by CVO from an image build of the `cluster-capi-operator`, and not from the mentioned `cluster-api-provider-x`.
+
+### User Stories
+
+As an administrator I want to _have a CAPI-on-OpenShift architecture as minimally complicated as possible_ so that I can _easily understand what is going on and debug potential issues on my cluster_.
+
+As an administrator I want to _have a CAPI-on-OpenShift architecture as minimally complicated as possible_ so that I can _extend my cluster by deploying a custom provider_.
+
+As an OpenShift engineer I want to _have a CAPI-on-OpenShift architecture as minimally complicated as possible_ so that I can _more easily maintain and extend it_.
+
+As an OpenShift engineer I want to _have a way to atomically apply a change to any provider_ so that I can avoid _payload breakages_.
+
+As an OpenShift engineer I want to _have a way to load and customize provider manifests before applying them_ so that I can _template the manifests payload with image references and other runtime tweaks_.
+
+As an OpenShift engineer I want to _have a way to load and customize provider manifests before applying them_ so that I can _future proof the system_ to be able to give users a way to selectively deploy one or more providers, even ones that are not matching the running platform.
+
+### Goals
+
+This proposal seeks to refactor OpenShift manifest generation to live in provider repositories & apply cloud platform information dynamically at runtime.  It will have succeeded when developers can update code and manifests for a Cluster API provider within one repository, and when users can selectively deploy providers at runtime.
+
+### Non-Goals
```

Comment:
We do not intend to support installing more providers than the platform on which the cluster resides

Reply:
Added to the non goals.